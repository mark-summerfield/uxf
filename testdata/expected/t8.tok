FileComment Str("A single comment may be placed at the start after the uxf header and\nbefore evertthing else.")
TClassBegin ttype=<Files>
  Field Str("Kind")
  Field Str("Filename")
TClassEnd
TClassBegin ttype=<Categories>
  Field Str("CID")
  Field Str("Title")
  Field Str("Selected")
TClassEnd
MapBegin
  Str("General")
  MapBegin
    Str("saved")
    Date(2022-03-21)
    Str("autosave")
    Bool(true)
    Str("historysize")
    Int(-35)
    Str("volume")
    Real(0.7)
    Str("files")
    TableBegin ttype=<Files>
      Str("current")
      Str("/home/mark/app/rs/tlm/PlaylistsTest.tlm")
      Str("recent1")
      Str("/home/mark/app/rs/tlm/PlaylistsTest.tlm")
      Str("recent2")
      Str("/home/mark/data/playlists-all.tlm")
    TableEnd
  MapEnd
  Str("Window")
  MapBegin
    Str("x")
    Int(383)
    Str("y")
    Int(124)
    Str("width")
    Int(590)
    Str("height")
    Int(536)
    Str("scale")
    Real(1.1)
  MapEnd
  Str("Magic")
  Bytes([31, 139])
  Str("Nested Dict")
  MapBegin
    Str("Classical")
    ListBegin
      Int(5)
      Bool(true)
    ListEnd
    Str("Modern Instrumental")
    ListBegin
      Int(4)
      Bool(false)
    ListEnd
    Str("New Acquistions")
    ListBegin
      Int(1)
      Bool(false)
    ListEnd
    Str("Nested List of Lists")
    ListBegin
      ListBegin
        Int(5)
        Str("Classical")
        Bool(true)
      ListEnd
      ListBegin
        Int(4)
        Str("Modern Instrumental")
        Bool(false)
      ListEnd
      ListBegin
        Int(1)
        Str("New Acquistions")
        Bool(false)
      ListEnd
    ListEnd
    Str("Nested Table")
    TableBegin ttype=<Categories>
      Int(5)
      Str("Classical")
      Bool(true)
      Int(4)
      Str("Modern Instrumental")
      Bool(false)
      Int(1)
      Str("New Acquistions")
      Bool(false)
      Int(2)
      Str("Pop")
      Bool(false)
      Int(3)
      Str("Punk")
      ?
      ?
      ?
      Bool(false)
      Int(6)
      Str("Unpopular Pop")
      Bool(false)
    TableEnd
  MapEnd
MapEnd
Eof
----------------------------------------
