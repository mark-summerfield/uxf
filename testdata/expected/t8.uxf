uxf 1.0 TLM Config
#<A single comment may be placed at the start after the uxf header and
before evertthing else.>
=Categories CID Title Selected
=Files Kind Filename
{
  <General>
    {
      <autosave> yes
      <files>
        (Files
          <current> </home/mark/app/rs/tlm/PlaylistsTest.tlm>
          <recent1> </home/mark/app/rs/tlm/PlaylistsTest.tlm>
          <recent2> </home/mark/data/playlists-all.tlm>
        )
      <historysize> -35
      <saved> 2022-03-21
      <volume> 0.7
    }
  <Magic> (:1F8B:)
  <Nested Dict>
    {
      <Classical> [5 yes]
      <Modern Instrumental> [4 no]
      <Nested List of Lists>
        [
          [5 <Classical> yes]
          [4 <Modern Instrumental> no]
          [1 <New Acquistions> no]
        ]
      <Nested Table>
        (Categories
          5 <Classical> yes
          4 <Modern Instrumental> no
          1 <New Acquistions> no
          2 <Pop> no
          3 <Punk> ?
          ? ? no
          6 <Unpopular Pop> no
        )
      <New Acquistions> [1 no]
    }
  <Window> {<height> 536 <scale> 1.1 <width> 590 <x> 383 <y> 124}
}
