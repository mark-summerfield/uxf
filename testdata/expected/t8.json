{
  "UXF^custom": "TLM Config",
  "UXF^comment": "A single comment may be placed at the start after the uxf header and\nbefore evertthing else.",
  "UXF^ttypes": [
    {
      "UXF^ttype": {
        "name": "Categories",
        "fields": {
          "CID": null,
          "Title": null,
          "Selected": null
        }
      }
    },
    {
      "UXF^ttype": {
        "name": "Files",
        "fields": {
          "Kind": null,
          "Filename": null
        }
      }
    }
  ],
  "UXF^data": {
    "General": {
      "autosave": true,
      "files": {
        "UXF^table": {
          "name": "Files",
          "comment": null,
          "fields": {
            "Kind": null,
            "Filename": null
          },
          "records": [
            [
              "current",
              "/home/mark/app/rs/tlm/PlaylistsTest.tlm"
            ],
            [
              "recent1",
              "/home/mark/app/rs/tlm/PlaylistsTest.tlm"
            ],
            [
              "recent2",
              "/home/mark/data/playlists-all.tlm"
            ]
          ]
        }
      },
      "historysize": -35,
      "saved": {
        "UXF^date": "2022-03-21"
      },
      "volume": 0.7
    },
    "Magic": {
      "UXF^bytes": "1F8B"
    },
    "Nested Dict": {
      "Classical": [
        5,
        true
      ],
      "Modern Instrumental": [
        4,
        false
      ],
      "Nested List of Lists": [
        [
          5,
          "Classical",
          true
        ],
        [
          4,
          "Modern Instrumental",
          false
        ],
        [
          1,
          "New Acquistions",
          false
        ]
      ],
      "Nested Table": {
        "UXF^table": {
          "name": "Categories",
          "comment": null,
          "fields": {
            "CID": null,
            "Title": null,
            "Selected": null
          },
          "records": [
            [
              5,
              "Classical",
              true
            ],
            [
              4,
              "Modern Instrumental",
              false
            ],
            [
              1,
              "New Acquistions",
              false
            ],
            [
              2,
              "Pop",
              false
            ],
            [
              3,
              "Punk",
              null
            ],
            [
              null,
              null,
              false
            ],
            [
              6,
              "Unpopular Pop",
              false
            ]
          ]
        }
      },
      "New Acquistions": [
        1,
        false
      ]
    },
    "Window": {
      "height": 536,
      "scale": 1.1,
      "width": 590,
      "x": 383,
      "y": 124
    }
  }
}